ruby_oop/mastermind/game.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require './computer_player'
+require './human_player'
+
+# class to play the mastermind game
+class Game
+  attr_accessor :human_player, :computer_player
+
+  AVAILABLE_COLORS = ['🟤', '🔴', '🔵', '🟢', '🟡', '🟣'].freeze
+  @secret_code = []
+
+  def initialize
+    @human_player = codebreaker_or_codemaker
+    @computer_player = !@human_player
+  end
+
+  def computer_codemaker
+    puts "Computer is selecting random colors...\n\n"
+    @secret_code = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@secret_code)
+  end
+
+  def computer_codebreaker
+    puts "Computer is selecting random colors...\n\n"
+    @guess_array = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@guess_array)
+  end
+
+  def human_codemaker
+    puts 'Enter a code to be broken: '
+    @secret_code = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+      HumanPlayer.new(@secret_code)
+  end
+
+  def human_codebreaker
+    @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                    pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    if @guess_array.include?(nil) || @guess_array.uniq.length != 4
+      puts 'Duplicate entries or empty entries not allowed. Enter a new selection again.'
+      @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    end
+    puts "\nYou guessed: #{@guess_array}"
+    HumanPlayer.new(@guess_array)
+  end
+
+  def codebreaker_or_codemaker
+    puts 'Do you want to be the codemaker or the codebreaker? Enter 1 for codemaker, 2 for codebreaker: '
+    choice = gets.chomp.to_i
+    case choice
+    when 1
+      puts 'You are the codemaker.'
+      true
+    when 2
+      puts 'You are the codebreaker.'
+      false
+    end
+  end
+
+  def color_index_match
+    @guess_array.each_with_index do |color, idx|
+      puts "#{color} is in the right spot!" if color.match(@secret_code[idx])
+    end
+  end
+
+  def color_found_in_array
+    @guess_array.each_with_index do |color, idx|
+      puts "Color #{color} is a part of the secret code." if @secret_code.include?(color)
+    end
+  end
+
+  # keep playing until there's a match OR 12 turns happen
+  def play
+    turn_number = 0
+    if @human_player
```

Comment:
Coming back to this variable name...what about something like `@player_codebreaker`.  In fact, looking at your `initialize` method again it seems like you don't need the `@computer_player` var.  I don't think it's being used elsewhere in the code. 🤔